### PR TITLE
Home screen link to monthly screening reports

### PR DESCRIPTION
**Pull request by siddh1004**
https://app.shortcut.com/simpledotorg/story/9801/integrate-the-monthly-screening-reports-button-s-visibility-with-the-server-boolean

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/home/patients/links/PatientsTabLinkEffect.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package org.simple.clinic.home.patients.links
+
+sealed class PatientsTabLinkEffect
```
Comment (olttwa) [inline app/src/main/java/org/simple/clinic/home/patients/links/PatientsTabLinkEffect.kt:3]: Why is it named PatientsTabLinkEffect and why does it need a method of `OpenPatientLineListDownloadDialog`?

File: app/src/main/java/org/simple/clinic/home/patients/links/PatientsTabLinkEvent.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+data class CurrentFacilityLoaded(
+    val facility: Facility
+) : PatientsTabLinkEvent()
+
+object MonthlyScreeningReportsClicked : PatientsTabLinkEvent() {
+  override val analyticsName = "Link: Monthly Screening Report clicked"
+}
+
+data class DownloadPatientLineListClicked(
```
Comment (olttwa) [inline app/src/main/java/org/simple/clinic/home/patients/links/PatientsTabLinkEvent.kt:20]: Does this mean `Download IHCI patient register` was clicked?
image

**Review by olttwa — CHANGES_REQUESTED** [review]

**Review by olttwa — APPROVED** [review]

File: app/src/main/java/org/simple/clinic/home/patients/links/PatientsTabLinkModel.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    val facility: Facility?,
+) : Parcelable {
+  companion object {
+    fun default() = PatientsTabLinkModel(
+        facility = null,
+    )
+  }
+
+  val monthlyScreeningReportsEnabled: Boolean
+    get() = facility?.config?.monthlyScreeningReportsEnabled == true
```
Comment (prabhanshuguptagit) [inline app/src/main/java/org/simple/clinic/home/patients/links/PatientsTabLinkModel.kt:18]: Should also check there is a questionnaire of type `monthly_screening_reports` in the DB.

Comment (siddh1004) [reply app/src/main/java/org/simple/clinic/home/patients/links/PatientsTabLinkModel.kt:18]: @prabhanshuguptagit will add in the separate PR. Thanks for the info.

**Pull request merged**